### PR TITLE
Add xml comments

### PR DESCRIPTION
**Pull request by Shane32**
See #1857 

Mostly putting finishing touches on namespaces/classes already nearly complete.

* src/GraphQL/Execution/DocumentExecuter.cs 
* src/GraphQL/Execution/ExecutionContext.cs 
* src/GraphQL/Execution/ExecutionResult.cs 
* src/GraphQL/Execution/GraphQLDocumentBuilder.cs 
* src/GraphQL/Instrumentation/FieldMiddlewareBuilder.cs 
* src/GraphQL/Instrumentation/FieldMiddlewareDelegate.cs 
* src/GraphQL/Instrumentation/IFieldMiddleware.cs 
* src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs 
* src/GraphQL/ResolveFieldContext/IResolveFieldContext.cs 
* src/GraphQL/ResolveFieldContext/ReadonlyResolveFieldContext.cs 
* src/GraphQL/ResolveFieldContext/ResolveConnectionContext.cs 
* src/GraphQL/ResolveFieldContext/ResolveFieldContext.cs 
* src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs 
* src/GraphQL/Resolvers/AsyncFieldResolver.cs 
* src/GraphQL/Resolvers/ExpressionFieldResolver.cs 
* src/GraphQL/Resolvers/FieldResolverExtensions.cs 
* src/GraphQL/Resolvers/FuncFieldResolver.cs 
* src/GraphQL/Resolvers/IFieldResolver.cs 
* src/GraphQL/Resolvers/NameFieldResolver.cs 
* src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs 
* src/GraphQL/Types/ISchema.cs

### CODE REVIEW

File: src/GraphQL/Execution/DocumentExecuter.cs [hunk 1]
```diff
@@ -317,13 +326,26 @@ await listener.AfterExecutionAsync(context)
             return context;
         }
 
+        /// <summary>
+        /// Returns the selected <see cref="Operation"/> given a specified <see cref="Document"/> and operation name.
+        /// <br/><br/>
+        /// Returns <c>null</c> if an operation cannot be found that matches the given criteria.
```
Comment (sungam3r) [inline src/GraphQL/Execution/DocumentExecuter.cs:332]:
```suggestion
        /// Returns <c>null</c> if an operation cannot be found that matches the given criteria.
        /// Returns the first operation from the document if no operation name was specified.
```

File: src/GraphQL/Execution/ExecutionResult.cs [hunk 2]
```diff
@@ -6,6 +6,9 @@
 
 namespace GraphQL
 {
+    /// <summary>
+    /// Represents the result of an execution
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionResult.cs:10]:
```suggestion
    /// Represents the result of an execution.
```

File: src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
+    /// </summary>
     public interface IResolveConnectionContext : IResolveFieldContext
     {
+        /// <summary>
+        /// Indicates if this connection only allows forward pagination requests.
+        /// </summary>
         bool IsUnidirectional { get; }
 
+        /// <summary>
+        /// For a forward pagination request, returns the maximum number of edges to be returned.
```
Comment (sungam3r) [inline src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs:18]: I have to admit that I haven't read about relay yet, but this comment and one below for `Last` seem odd.

File: src/GraphQL/ResolveFieldContext/ResolveFieldContext.cs [hunk 4]
```diff
@@ -13,54 +13,79 @@ namespace GraphQL
...
         public IDictionary<string, Field> SubFields { get; set; }
 
+        /// <inheritdoc/>
         public IServiceProvider RequestServices { get; set; }
 
+        /// <inheritdoc/>
         public IDictionary<string, object> Extensions { get; set; }
 
+        /// <summary>
+        /// Initialize an instance with all fields set to their default values.
```
Comment (sungam3r) [inline src/GraphQL/ResolveFieldContext/ResolveFieldContext.cs:83]:
```suggestion
        /// Initializes a new instance with all fields set to their default values.
```

File: src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs (hunk 3 above)
Comment (Shane32) [reply src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs:18]: I paraphrased from the spec. See “arguments” here:

https://relay.dev/graphql/connections.htm

Probably not the best description but that’s the best I could come up with.

File: src/GraphQL/Resolvers/ExpressionFieldResolver.cs [hunk 5]
```diff
@@ -4,17 +4,22 @@
...
     /// <summary>
-    /// Returns a value from the field's graph type's source object, based on a predefined expression
+    /// Returns a value from the field's graph type's source object, based on a predefined expression.
     /// </summary>
     public class ExpressionFieldResolver<TSourceType, TProperty> : IFieldResolver<TProperty>
     {
         private readonly Func<TSourceType, TProperty> _property;
 
+        /// <summary>
+        /// Initializes a new instance that runs the specified expression.
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/ExpressionFieldResolver.cs:14]:
```suggestion
        /// Initializes a new instance that runs the specified expression when resolving a field.
```

File: src/GraphQL/Resolvers/FuncFieldResolver.cs [hunk 6]
```diff
@@ -9,30 +9,36 @@ public class FuncFieldResolver<TReturnType> : IFieldResolver<TReturnType>
     {
         private readonly Func<IResolveFieldContext, TReturnType> _resolver;
 
+        /// <summary>
+        /// Initializes a new instance that runs the specified delegate.
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/FuncFieldResolver.cs:13]:
```suggestion
        /// Initializes a new instance that runs the specified delegate when resolving a field..
```

File: src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs [hunk 7]
```diff
@@ -32,6 +32,9 @@ public AutoRegisteringObjectGraphType(params Expression<Func<TSourceType, object
             AutoRegisteringHelper.SetFields(this, GetRegisteredProperties(), excludedProperties);
         }
 
+        /// <summary>
+        /// Returns a list of proprties that should have fields created for them.
```
Comment (sungam3r) [inline src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs:36]:
```suggestion
        /// Returns a list of properties that should have fields created for them.
```

File: src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs [hunk 8]
```diff
@@ -63,6 +66,9 @@ public AutoRegisteringInputObjectGraphType(params Expression<Func<TSourceType, o
...
```
Comment (sungam3r) [inline src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs:70]:
```suggestion
        /// Returns a list of properties that should have fields created for them.
```

File: src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs (hunk 3 above)
Comment (sungam3r) [reply src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs:18]: From the third reading, I roughly understood what it was about. Let's give each property and class a link to the relevant paragraphs of the specification.

Comment (Shane32) [reply src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs:18]: Perhaps I misunderstood you, but we don't have embedded links to the graphql specification anywhere else in the xml documentation.  The xml comments assume a basic understanding of the overall functionality, and are written as such (just like the rest of the xml comments).  Further, some of the properties have no bearing on the actual spec, and probably shouldn't even exist.  For instance, `IsUnidirectional` is only a placeholder to hold temporary data within the program.

I'm leaving it as-is for now.  If we need more overall documentation for relay types, we should add it to the https://graphql-dotnet.github.io/docs/getting-started/relay page.  (As the page is nearly empty right now!)  I did add a link to the spec on that page.

Comment (sungam3r) [reply src/GraphQL/ResolveFieldContext/IResolveConnectionContext.cs:18]: OK. In general, sooner or later, I expected to remove all these relay types in favor of a separate relay project.

**Review by sungam3r — APPROVED** [review]

**Comment by Shane32** [comment]
Either there's a temporary problem with GitHub Actions, or the `Labeler` task doesn't work with pull requests from forks.

**Comment by sungam3r** [comment]
It's a known issue. Try to create branch from origin repo.

**Comment by Shane32** [comment]
If labeler doesn’t work let’s dump it. It’s pretty useless anyway.

**Comment by sungam3r** [comment]
It automates labeling of our own PRs.

**Comment by Shane32** [comment]
I've been watching it since it was installed.  I don't think it's useful.

**Comment by sungam3r** [comment]
Dependabot sets the labels to my and your PRs. Taking into account that we are doing the bulk of the PR recently, I would not say that it is useless.

**Pull request merged**